system-int-test/test/cmr/system_int_test/ingest/generics_test.clj

```diff
@@ -251,10 +252,132 @@
     (is (= result1 {:concept-id concept-id, :revision-id 2,:warnings nil, :existing-errors nil}))
     (is (= result2 {:errors ["CollectionDraft with native id [NativeId] in provider [PROV1] does not exist."]}))
-
+    
     ;;Verify that searching for the concept from elastic search won't return anything after the delete.
     (let [result (search-generics-test/search-request "collection-drafts" (str "concept_id=" concept-id))
           status (:status result)
           body (:body result)]
       (is (= status 200))
       (is (string/includes? body "<hits>0</hits>")))))
+
+;; Test that collection-draft can be published
+;; and the draft is removed from metadatadb.
+(deftest test-collection-draft-publishing
+  ;; Drafts have permissions to ingest on PROV1, but not on PROV2.
+  (let [;; Ingest a collection draft
+        cd-native-id "CD-NativeId"
+        coll-native-id "COLL-NativeId"
+        coll-draft (gen-util/ingest-generic-document
+                    nil "PROV1" cd-native-id :collection-draft gen-util/collection-draft :post)
+        cd-concept-id (:concept-id coll-draft)
+        coll-published (ingest/publish-non-variable-draft
+                        cd-concept-id coll-native-id nil)
+        coll-concept-id (:concept-id coll-published)
+
+        _ (index/wait-until-indexed)
+        ;; search for the published collection from elastic search should return result.
+        search-coll-result (search-generics-test/search-request "collections" (str "concept_id=" coll-concept-id))
+        search-coll-status (:status search-coll-result)
+        search-coll-body (:body search-coll-result)
+
+        ;; search for the draft from elastic search should return nothing.
+        search-draft-result (search-generics-test/search-request "collection-drafts" (str "concept_id=" cd-concept-id))
+        search-draft-status (:status search-draft-result)
+        search-draft-body (:body search-draft-result)
+       
+        ;; deleting the draft should fail. 
+        draft-delete-result (gen-util/ingest-generic-document
+                             nil "PROV1" cd-native-id :collection-draft gen-util/collection-draft :delete)]
+    ;;Verify that searching for the published collection from elastic search does return result.
+    (is (= search-coll-status 200))
+    (is (string/includes? search-coll-body "<hits>1</hits>"))
+
+    ;;Verify that searching for the draft from elastic search returns nothing
+    (is (= search-draft-status 200))
+    (is (string/includes? search-draft-body "<hits>0</hits>"))
+ 
+    ;;The draft delete result should indicate the draft does not exist. 
+    (is (= draft-delete-result {:errors ["CollectionDraft with native id [CD-NativeId] in provider [PROV1] does not exist."]}))))
+
+;; Test that generic-doc draft can be published
+;; and the draft is removed from metadatadb.
+(deftest test-generic-draft-publishing
```

Comment:
It definitely makes sense to me to test the generics specifically but, for my own understanding, in this publishing feature, they are getting parsed with the same code the collection, service etc?

Reply:
No. collection, service, tool are going through the same code. but generic-docs are different.